include/pybind11/pybind11.h

```diff
@@ -1435,29 +1445,22 @@ struct enum_base {
         auto static_property = handle((PyObject *) get_internals().static_property_type);
 
         m_base.attr("__repr__") = cpp_function(
-            [](handle arg) -> str {
+            [](object arg) -> str {
```

Comment:
Why handle to object? You don't need reference counting here, do you?

Reply:
Argh, good catch. I should've look more closely at the details rather than eyeballing the structural changes and shrugging "OK"...

Reply:
OK, so I just tried this out, and the reason seems to be because there's no `int_` conversion from `handle` (you can `reinterpret_steal`/`reinterpret_borrow`, but those will not do the `int(...)` conversion).

Reply:
See e.g. `__getstate__` and `__hash__`, also doing that. I don't immediately see a better solution, then? (though if feels there ought to be one, changing the interface of the Python types; but that's a different discussion/PR)

Reply:
Maybe pass the `object` by reference, to avoid reference counting?

Reply:
@bstaletic Hmmm, interesting.

Thinking about it a bit futher: will it, though? I believe (though the `cast`, `make_caster`, `cast_op`, `PYBIND11_TYPE_CASTER`, ... code is quite convoluted) that the `object` will be created and moved from the caster anyway, if you take it by value. So it shouldn't matter?

Reply:
I vote to keep it this way, merge this PR, and create an issue/discussion on why you can't call the `int_` conversion on a handle (like you can for `str`).

Reply:
I'm okay with that.